Utils/src/CandidateTrackFilter.cc

```diff
@@ -0,0 +1,380 @@
+class TrackInfos {
+public:
+	TrackInfos() {
+		trks                         = std::make_unique<vector<math::XYZVector>>();
+		trks_referencepoint          = std::make_unique<vector<math::XYZPoint>>();
+		trks_chg                     = std::make_unique<vector<int>>();
+		trks_dzpv                    = std::make_unique<vector<double>>();
+		trks_dzerrorpv               = std::make_unique<vector<double>>();
+		trks_dxypv                   = std::make_unique<vector<double>>();
+		trks_dxyerrorpv              = std::make_unique<vector<double>>();
+		trks_normalizedchi2          = std::make_unique<vector<double>>();
+		trks_pterror                 = std::make_unique<vector<double>>();
+		trks_etaerror                = std::make_unique<vector<double>>();
+		trks_phierror                = std::make_unique<vector<double>>();
+		trks_qoverperror             = std::make_unique<vector<double>>();
+		trks_ip2d                    = std::make_unique<vector<double>>();
+		trks_ip2dsig                 = std::make_unique<vector<double>>();
+		trks_ip3d                    = std::make_unique<vector<double>>();
+		trks_ip3dsig                 = std::make_unique<vector<double>>();
+		trks_found                   = std::make_unique<vector<int>>();
+		trks_lost                    = std::make_unique<vector<int>>();
+		trks_quality                 = std::make_unique<vector<int>>();
+		trks_hitpattern              = std::make_unique<vector<vector<int>>>();
+		trks_matchedtopfcand         = std::make_unique<vector<bool>>();
+		pfcands_numberofhits         = std::make_unique<vector<int>>();
+		pfcands_numberofpixelhits    = std::make_unique<vector<int>>();
+		pfcands_firsthit             = std::make_unique<vector<int>>();
+		pfcands_frompv               = std::make_unique<vector<int>>();
+		pfcands_pvassociationquality = std::make_unique<vector<int>>();
+		pfcands_dzassociatedpv       = std::make_unique<vector<double>>();
+	}
+
+	void fill(const reco::Vertex & primaryVertex, const pat::PackedCandidate & pfCand, const reco::Track & track, const reco::TransientTrack & transientTrack, bool trackMatch) {
+		// basic information from the reco::Track
+		trks->push_back(track.momentum());
+		trks_referencepoint->push_back(track.referencePoint());
+		trks_chg->push_back(track.charge());
+		trks_dzpv->push_back(track.dz()); //returns the ip wrt PV[0] 
+		trks_dzerrorpv->push_back(track.dzError());
+		trks_dxypv->push_back(track.dxy());
+		trks_dxyerrorpv->push_back(track.dxyError());
+		trks_normalizedchi2->push_back(track.normalizedChi2());
+		trks_pterror->push_back(track.ptError());
+		trks_etaerror->push_back(track.etaError());
+		trks_phierror->push_back(track.phiError());
+		trks_qoverperror->push_back(track.qoverpError());
+		trks_found->push_back(track.found());
+		trks_lost->push_back(track.lost());
+		// track quality enum options: http://cmsdoxygen.web.cern.ch/cmsdoxygen/CMSSW_10_2_11_patch1/doc/html/d8/df2/classreco_1_1TrackBase.html#aeee12ec6a3ea0d65caa2695c84ab25d5
```

Comment:
would prefer to replace doxygen links w/ github links

Reply:
Done